src/mpl/src/misc/mpl_mkstemp.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpl.h"
+
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+/* here to prevent "has no symbols" warnings from ranlib on OS X */
+static int dummy ATTRIBUTE((unused)) MPL_USED = 0;
+
+#ifndef MPL_HAVE_MKSTEMP
+
+#warning The mkstemp function is not provided in the standard library.
+#warning Using a possibly insecure replacement function.
+
+#if !defined (HAVE_RAND) || !HAVE_RAND
+static notrand = 0;
+#define srand(x) notrand = x
+#define rand() (++notrand)
+#endif
```

Comment:
`rand` is in standard C. How about remove the fallback and let's assume we always will have them?